### ISSUE TITLE
Date field left null in detection while updating techniques

### ISSUE THREAD

**Issue by AlphaCorvus** [siriussecurity/dettectinator#5]
Hi there 👋,

Following #4, I am creating a techniques admin from DeTT&CK using `--yaml --yaml-all-techniques`. This way, I have all the techniques referenced and empty.

I am now able to use dettectinator to update the technique admin file but, I get errors when using DeTT&CK to generate a layer of detection. The error I have is the following:

```
dettect.py d -ft techniques_dash.yaml -l
[!] Possible error in YAML file at: T1018. Error: '>' not supported between instances of 'datetime.date' and 'NoneType'
```

The reason may that as the detection is created with null dates, it is not replaced by dettectinator. 

<details>
  <summary>Before the update by dettectinator</summary>
  
```
- technique_id: T1018
  technique_name: Remote System Discovery
  detection:
    applicable_to:
    - all
    location:
    - ''
    comment: ''
    score_logbook:
    - date: 
      score: -1
      comment: ''
  visibility:
  - applicable_to:
    - default
    comment: ''
    score_logbook:
    - date: 2023-01-30 16:42:10.026048
      score: 0
      comment: ''
      auto_generated: true
  - applicable_to:
    - Windows
    comment: ''
    score_logbook:
    - date: 2023-01-30 16:42:10.026048
      score: 3
      comment: ''
      auto_generated: true
```
</details>

<details>
  <summary>After an update by dettectinator</summary>
  
```
- technique_id: T1018
    technique_name: Remote System Discovery
    detection:
      - applicable_to:
          - all
        location:
          - ''
          - 'Rule1
          - 'Rule2'
        comment: ''
        score_logbook:
          - date: null
            score: -1
            comment: ''
          - date: 2023-01-30T00:00:00.000Z
            score: -1
            comment: 'Auto added by Dettectinator. TODO: Check score. Detection rule added: Rule1, Rule2'
    visibility:
      - applicable_to:
          - default
        comment: ''
        score_logbook:
          - date: 2023-01-30 16:42:10.026048
            score: 0
            comment: ''
            auto_generated: true
      - applicable_to:
          - Windows
        comment: ''
        score_logbook:
          - date: 2023-01-30 16:42:10.026048
            score: 3
            comment: ''
            auto_generated: true
```
  
</details>

Note: An empty value is left before Rule1 and Rule2, not sure if this will create additional issues.

The import seem successful and the health of the YAML is fine. Do you observe the same issue? I should have checked before validate the issue #4, sorry.

**Comment by rubinatorz** [comment]
hi @AlphaCorvus,

This was a problem in DeTT&CT and I just fixed it. Can you please pull the latest changes from the DeTT&CT repo and try to generate the YAML again. And then use dettectinator to update en use DeTT&CT to generate the json detection layer?

Regards,
Ruben

**Comment by AlphaCorvus** [comment]
Hi @rubinatorz,

Many thanks for fixing this. The date issue is resolved, I can generate a layer. However, when adding a new detection, dettectinator does not add a score of 1 in the `score_logbook:`. Is something expected?

```yaml
score_logbook:
          - date: 2023-01-31 14:14:32.846103
            score: -1
            comment: ''
          - date: 2023-01-31T00:00:00.000Z
            score: -1
            comment: 'Auto added by Dettectinator. TODO: Check score. Detection rule added: Rule1. Detection rule added: Rule2'
```

Regards,
Mathieu

**Comment by rubinatorz** [comment]
hi @AlphaCorvus,

Just committed a fix for dettectinator that solves this. The last change I did in DeTT&CT resulted in this "side effect".

If you have anything else, please let me know!

Regards,
Ruben

**Comment by AlphaCorvus** [comment]
Hi @rubinatorz,

I am so sorry for keeping asking questions/requests but, when I use `dettect.py d -ft techniques_dash.yaml -l`, the layer is empty. When I remove the following part from the `score_logbook:`:

```
          - date: 2023-01-31 14:14:32.846103
            score: -1
            comment: ''
```

...It finds the detection. I also tried to set the date of the log added by dettectinator to `2023-01-31T16:00:00.000Z` but it didn't solve the issue.

Regards,
Mathieu

**Comment by rubinatorz** [comment]
hi @AlphaCorvus 

No worries, happy to help to make DeTT&CT and dettectinator better!

I reverted one of my changes in DeTT&CT... the one where I set the default date for empty logbook entries to today's date. That resulted in too much issues elsewhere, like your last issue.

Then I added a fix in dettectinator to solve your initial `Error: '>' not supported between instances of 'datetime.date' and 'NoneType'` issue. When adding a score with dettectinator I first check if there's a default entry with no date and score -1; if so, that entry will be deleted.

Let me know if this helps.
